activerecord/lib/active_record/counter_cache.rb

```diff
@@ -112,14 +155,36 @@ def increment_counter(counter_name, id)
       #
       # * +counter_name+ - The name of the field that should be decremented.
       # * +id+ - The id of the object that should be decremented or an array of ids.
+      # * +options+ - {touch: true} will update the `timestamp_attributes_for_update_in_model` column.
+      #               {touch: :last_discussed_at} will update the `last_discussed_at` column.
+      #               {touch: %i( updated_at last_discussed_at )} will update the `updated_at` and `last_discussed_at` columns.
       #
       # ==== Examples
       #
       #   # Decrement the posts_count column for the record with an id of 5
       #   DiscussionBoard.decrement_counter(:posts_count, 5)
-      def decrement_counter(counter_name, id)
-        update_counters(id, counter_name => -1)
+      #
+      #   # Decrement the posts_count column for the record with an id of 5
+      #   # and update the updated_at value.
+      #   DiscussionBoard.decrement_counter(:posts_count, 5, updated_at: true)
+      def decrement_counter(counter_name, id, options={})
+        update_counters(id, { counter_name => -1 }.merge(options))
       end
+
+      private
+
+        def resolve_timestamp_columns(object, touch)
+          case touch
+          when ::TrueClass
+            object.send(:timestamp_attributes_for_update_in_model)
+          when ::Symbol, ::Array
+            touch
+          when ::NilClass
+            nil
+          else
+            raise ArgumentError, "Expected `true`, a symbol, or an array of symbols, got #{touch}"
+          end
```

Comment:
I'd just go with:

```ruby
def resolve_timestamp_columns(object, touch)
  if touch == true
    object.timestamp_attributes_for_update_in_model
  else
    Array(touch)
  end
end
```

Don't think we need to be that defensive here and add an exception.

I also dislike the `send` part. I think we should make the method public but mark it as `# :nodoc:`. That way it's still private API, but has public visibility.